### ISSUE TITLE
pass the JS event onCopy()

### ISSUE THREAD

**Issue by PierBover** [nkbt/react-copy-to-clipboard#23]
So I want to do something on the DOM element that originated the copy.

I've tried using `onClick` but nothing happens. I will assume because the handler gets overwritten by the `CopyToClipboard` internal logic.

So I tried using 'onCopy', but only the copied text gets passed to the handler.

It would be great if the DOM element where the click was originated was passed as a second argument of `onCopy` or even better pass the full JS event.

Is there any logical argument against this?

@nkbt  would you accept a pull request with this functionality?

**Comment by jseminck** [comment]
We use the Clipboard button like this:

```js
return (
            <ClipboardButton
                data-clipboard-text={this.props.data}
                onSuccess={::this.onSuccess}
                onError={::this.onError}
                component="div"
            >
                <a onClick={::this.onClick} >
                    Copy me !
                </a>
            </ClipboardButton>
        );
```

Where the onClick triggers some other stuff...

Would this work for you?

**Comment by nkbt** [comment]
@PierBover ah, right! Of course, original onClick should not be overridden.

Should be pretty much like here: https://github.com/nkbt/react-debounce-input/blob/master/src/Component.js#L140-L148

**Comment by PierBover** [comment]
Hey @jseminck what's this `::` notation in `onClick={::this.onClick}`?

@nkbt yes, that would make more sense :)

If I have time later I will create a PR.

**Comment by nkbt** [comment]
@PierBover `::` is auto-binding to `this`. Some very early ES proposal (dropped, I reckon)

**Comment by PierBover** [comment]
The `::` syntax is a bit ugly to be honest, but we do really need some syntactic sugar for that.

**Comment by nkbt** [comment]
@PierBover yeah it totally is. But I think we really don't need anything with `this` in JS at all ;).

**Comment by PierBover** [comment]
LOL I doubt we will ever get rid of JS _thisitis_ :)

If [ES4 had been approved 10 years ago ](https://www.quora.com/Why-was-ECMAScript-4-ES4-never-implemented-outside-ActionScript-3-at-Adobe) we'd be dealing with a much better language today.

**Comment by nkbt** [comment]
I actually fixed this with https://github.com/nkbt/react-copy-to-clipboard/pull/27

**Comment by nkbt** [comment]
Published react-copy-to-clipboard@4.1.2 with this fix.
